ElectronClient/gui/NoteText.jsx

```diff
@@ -1475,12 +1480,27 @@ class NoteTextComponent extends React.Component {
 
 			// Add the number of newlines to the row
 			// and add the length of the final line to the column (for strings with no newlines this is the string length)
-			const newRange = {
-				start: { row: r.start.row + str1Split.length - 1,
-					column: r.start.column + str1Split[str1Split.length - 1].length },
-				end: { row: r.end.row + str1Split.length - 1,
-					column: r.end.column + str1Split[str1Split.length - 1].length },
-			};
+
+			let newRange = {};
+
+
+
+			if (!byLine) {
+				// Correcting selection offset when text is surrounded by white spaces
```

Comment:
I'm not sure the `!byLine` means surrounded by whitespace.  I thought it means that the wrapping will happen around all the text selected (as opposed to wrap each line in the selection).  If I am right, is it the comment or the code that is incorrect?

Reply:
> in this PR #2582 we added the functionality to convert text into bold/italics/code even though they were surrounded by spaces, so in wrapSelection we added this condition that is if byline==false (which means treat the selected text as a block not inline) implement this rule
> 
> <img alt="byline" width="848" src="https://user-images.githubusercontent.com/54576074/76683509-9e606b00-662a-11ea-8402-147ab055baa2.png">
> 
> Now the issue of **text not getting properly selected text inside markdown rule** (REASON: _because it counts the white spaces which creates an offset in the selected text, it should ignore all the white spaces before and after the markdown rule and select only the text inside the rule_) was created when you implemented this rule(_**the one in the above image**_),
> 
> So only cases with byLine== false need the code for correction in selection offset, Hope this clears it up!

Reply:
The comment is a bit misleading because It is not "correcting..." at that point because it hasn't been started yet. It is correct in the context of your changes, but not in the big picture.

Also, I'm pretty sure that `byLine` differentiates between wrapping each line of selected text, versus wrapping the whole of the selected text.  If it is about `inline` and `block` mode, then the argument should have a different name.  Just take care around that, because maybe it isn't doing what you think it is.

PS code screenshots are not appreciated in conversations unless special circumstances. Please use code blocks.